2018-11-03  Jordan Combitsis  <combitsi@lnxsrv09.seas.ucla.edu>

	* timespec.h (timespec_cmp): Use conditional expression(s) instead of
	subtraction in order to avoid integer overflow.

--- lib/timespec.h	2005-09-21 23:05:39.000000000 -0700
+++ lib/timespec.h	2018-11-03 18:23:37.218908000 -0700
@@ -45,8 +45,7 @@
 static inline int
 timespec_cmp (struct timespec a, struct timespec b)
 {
-  int diff = a.tv_sec - b.tv_sec;
-  return diff ? diff : a.tv_nsec - b.tv_nsec;
+  return a.tv_nsec > b.tv_nsec ? 1 : (a.tv_nsec < b.tv_nsec ? -1 : a.tv_nsec - b.tv_nsec);
 }
 
 # if ! HAVE_DECL_NANOSLEEP
